test/utils/poll-helper.ts

```diff
@@ -0,0 +1,127 @@
+import { expect } from "chai";
+import fetchMock from "fetch-mock";
+import pMap from "p-map";
+import * as sinon from "sinon";
+
+import { HTTPMethod } from "../../src/api/RestAPI.js";
+import { POLLING_INTERVAL, POLLING_TIMEOUT } from "../../src/common/constants.js";
+import { TimeoutError } from "../../src/errors/TimeoutError.js";
+
+const tickPoll = async (sandbox: sinon.SinonSandbox, ticks = 3) =>
+    pMap(Array(ticks).fill(0), async () => sandbox.clock.tickAsync(POLLING_INTERVAL), { concurrency: 1 });
+
+const timeoutPoll = async (sandbox: sinon.SinonSandbox) => sandbox.clock.tickAsync(POLLING_TIMEOUT);
+
+export const assertPoll = async <Item>(
+    pathMatcher: fetchMock.MockMatcher,
+    call: () => Promise<Item>,
+    sandbox: sinon.SinonSandbox,
+    successItem: Item,
+    pendingItem: Item | Item[]
+) => {
+    const pendingItems = Array.isArray(pendingItem) ? pendingItem : [pendingItem];
```

Comment:
maybe use [`arrify`](https://www.npmjs.com/package/arrify) here?

Reply:
Sounds good to me. Thanks! 🙂